src/components/Navbar.tsx

```diff
@@ -17,24 +21,235 @@ const ConnectButton = styled(Button)`
             <h4>
                 <Link to="/">thechat.eth</Link>
             </h4>
             {account ? (
-                <AddressButton
-                    type="button"
-                    onClick={disconnect}
-                    address={account}
-                />
+                <>
+                    <AddressButton
+                        type="button"
+                        onClick={openModal}
+                        address={account}
+                    />
+                    <ReactModal
+                        isOpen={modalIsOpen}
+                        contentLabel="Connect a wallet"
+                        style={customStyles}
+                    >
+                        <StyledModalContent>
+                            <ModalHeader>
+                                <h2>Account</h2>
+                                <CloseButton onClick={closeModal}>
+                                    <svg
+                                        width="16"
+                                        height="16"
+                                        viewBox="0 0 16 16"
+                                        fill="none"
+                                        xmlns="http://www.w3.org/2000/svg"
+                                    >
+                                        <path
+                                            d="M15 1L0.999999 15"
+                                            stroke="#59799C"
+                                            stroke-width="1.5"
+                                            stroke-linecap="round"
+                                        />
+                                        <path
+                                            d="M1 1L15 15"
+                                            stroke="#59799C"
+                                            stroke-width="1.5"
+                                            stroke-linecap="round"
+                                        />
+                                    </svg>
+                                </CloseButton>
+                            </ModalHeader>
+                            <ContentContainer>
+                                <ContentHeader>
+                                    <Subheading>
+                                        Connected with Metamask
+                                    </Subheading>
+                                    <ChangeButton>Change</ChangeButton>
+                                </ContentHeader>
+                                <input
+                                    disabled={true}
+                                    placeholder={trunc(account)}
+                                />
+                                <LinkContainer>
+                                    <ExternalLink
+                                        href={`https://etherscan.io/address/${account}`}
+                                        target="_blank"
+                                        rel="noreferrer"
```

Comment:
Let's have both `noreferrer` and `noopener` in here. Some browsers need both to act right.